linodecli/configuration/__init__.py

```diff
@@ -242,6 +242,14 @@ def update(
                 continue
             if self.config.has_option(username, key):
                 value = self.config.get(username, key)
+            # different types of database creation use different endpoints,
+            # so we need to set the default engine value based on the type
+            elif key == "engine":
+                if action == "mysql-create" and self.config.has_option(username, "mysql_engine"):
```

Comment:
This looks really good! 
Maybe in the future we could optionally accept `x-linode-cli-allowed-defaults` as a dict so config default mappings could be handled in the spec.

e.g.
```yaml
x-linode-cli-allowed-defaults:
    engine: mysql_engine
```

Reply:
Oh good idea. I didn't realize it can work this way. Let me create a follow-up ticket to optimize in the future.